homeassistant/components/image_processing/tensorflow.py

```diff
@@ -0,0 +1,346 @@
+        entities.append(TensorFlowImageProcessor(
+            hass, camera[CONF_ENTITY_ID], camera.get(CONF_NAME),
+            session, detection_graph, category_index, config))
+
+    add_entities(entities)
+
+
+class TensorFlowImageProcessor(ImageProcessingEntity):
+    """Representation of an TensorFlow image processor."""
+
+    def __init__(self, hass, camera_entity, name, session, detection_graph,
+                 category_index, config):
+        """Initialize the TensorFlow entity."""
+        model_config = config.get(CONF_MODEL)
+        self.hass = hass
+        self._camera_entity = camera_entity
+        if name:
+            self._name = name
+        else:
+            self._name = "TensorFlow {0}".format(
+                split_entity_id(camera_entity)[1])
+        self._session = session
+        self._graph = detection_graph
+        self._category_index = category_index
+        self._min_confidence = config.get(CONF_CONFIDENCE)
+        self._file_out = config.get(CONF_FILE_OUT)
+
+        # handle categories and specific detection areas
+        categories = model_config.get(CONF_CATEGORIES)
+        self._include_categories = []
+        self._category_areas = {}
+        for category in categories:
+            if isinstance(category, dict):
+                category_name = category.get(CONF_CATEGORY)
+                category_area = category.get(CONF_AREA)
+                self._include_categories.append(category_name)
+                self._category_areas[category_name] = [0, 0, 1, 1]
+                if category_area:
+                    self._category_areas[category_name] = [
+                        category_area.get(CONF_TOP),
+                        category_area.get(CONF_LEFT),
+                        category_area.get(CONF_BOTTOM),
+                        category_area.get(CONF_RIGHT)
+                    ]
+            else:
+                self._include_categories.append(category)
+                self._category_areas[category] = [0, 0, 1, 1]
+
+        # Handle global detection area
+        self._area = [0, 0, 1, 1]
+        area_config = model_config.get(CONF_AREA)
+        if area_config:
+            self._area = [
+                area_config.get(CONF_TOP),
+                area_config.get(CONF_LEFT),
+                area_config.get(CONF_BOTTOM),
+                area_config.get(CONF_RIGHT)
+            ]
+
+        template.attach(hass, self._file_out)
+
+        self._matches = {}
+        self._total_matches = 0
+        self._last_image = None
+
+    @property
+    def camera_entity(self):
+        """Return camera entity id from process pictures."""
+        return self._camera_entity
+
+    @property
+    def name(self):
+        """Return the name of the image processor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        return self._total_matches
+
+    @property
+    def state_attributes(self):
```

Comment:
Should be `device_state_attributes`, `state_attributes` is reserved for entity base classes.